hastexo/hastexo.py

```diff
@@ -375,12 +375,15 @@ def add_xml_to_node(self, node):
             if self.providers:
                 for provider in self.providers:
                     provider_node = etree.SubElement(root, 'provider')
-                    provider_template = provider.get("template", None)
-                    if provider_template in (None, "None"):
-                        provider.pop("template")
-                    provider_environment = provider.get("environment", None)
-                    if provider_environment in (None, "None"):
-                        provider.pop("environment")
+                    if 'template' in provider:
```

Comment:
Maybe you want to add some of the information as to _why_ you're doing it this way, as a comment? That would be a nice touch for Future Maari who comes back at this in a year or two, and scratches her head why the extra check is there. :)

Reply:
Sounds good, added a comment.